src/Pelagos/Util/DOIutil.php

```diff
@@ -319,42 +319,47 @@ public function getDOIMetadata($doi)
      * @throws HttpClientErrorException When there was an 4xx error negotiating with EZID.
      * @throws HttpServerErrorException When there was an 5xx error negotiating with EZID.
      *
-     * @return array
+     * @return boolean
      */
     public function deleteDOI($doi)
     {
-        // Add doi: to doi is it doesn't exist.
-        $doi = preg_replace('/^(?:doi:)?(10.\S+)/', 'doi:$1', $doi);
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, $this->url . '/id/' . $doi);
-        curl_setopt($ch, CURLOPT_USERPWD, $this->doiusername . ':' . $this->doipassword);
-        curl_setopt($ch, CURLOPT_POST, true);
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
-        $output = curl_exec($ch);
-
-        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
-        curl_close($ch);
-
-        //check to see if it worked.
-        if (200 != $httpCode) {
-            $expMsg = "ezid failed with:$httpCode($output)";
-            if ($httpCode >= 400 and $httpCode <= 499) {
-                throw new HttpClientErrorException($expMsg, $httpCode);
-            } elseif ($httpCode >= 500 or $httpCode == 0) {
-                throw new HttpServerErrorException($expMsg, $httpCode);
-            }
-        }
-
-        $metadata = array();
-        foreach (explode("\n", $output) as $line) {
-            $split = preg_split('/:/', $line, 2);
-            if (count($split) > 1) {
-                $metadata[$split[0]] = trim($split[1]);
+        $doiMetadata = $this->getDOIMetadata($doi);
+        if ($doiMetadata['_status'] === 'reserved') {
```

Comment:
This check is to determine, if the doi in Datacite is in Draft (in EZ id terms, 'reserved')

Reply:
This works, just looking for shortcuts, but there aren't any.